tgui/packages/tgui/interfaces/Modularshieldgen.tsx

```diff
@@ -0,0 +1,147 @@
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { Stack, Section, ProgressBar, Button, NumberInput } from '../components';
+
+type Modularshieldgendata = {
+  max_strength: number;
+  current_strength: number;
+  max_regeneration: number;
+  current_regeneration: number;
+  max_radius: number;
+  current_radius: number;
+  active: Boolean;
+  recovering: Boolean;
+  exterior_only: Boolean;
+};
+
+export const Modularshieldgen = (props, context) => {
```

Comment:
keeping in line with JS stuff, `modularShieldGenerator`.

Reply:
it says my ui is missing an export when i change that

Reply:
sorry i meant `ModularShieldGen` in this instance (edit: there's another part of the review that i think i found the issue with?)

Reply:
i tried changing both, when i changed both to camel case it said there was no "modularShieldGen" thingy so im assuming it has to do with the name of the file